### DISCUSSION TITLE
swin_unetr_btcv_segmentation

### DISCUSSION THREAD

**Discussion by katy070222** [Project-MONAI/tutorials#1204] [Q&A] [answered]
hi, I want to change this model and let it can work in 1 channel input & 2 channel output.(background&spleen).
That is,change swin_unetr_btcv_segmentation to work like spleen_ct_segmentation.

I change it like below.But,it didn't work. it would stop and keep"2023-02-03 04:05:35,605 - ignite.engine.engine.SupervisedTrainer - INFO - Engine run resuming from iteration 0, epoch 0 until 2 epochs",and  have't any  Error message.

![image](https://user-images.githubusercontent.com/119793057/216438906-a564ca41-8921-4927-9a5f-88f680e26a87.png)

I check the place which it jump in and don't jump out. The place is in "/home/u2562867/.local/lib/python3.8/site-packages/ignite/engine/engine.py",1064 line =" self.state.output = self._process_function(self, self.state.batch)",which is also in "/home/u2562867/.local/lib/python3.8/site-packages/monai/engines/trainer.py",226 line"engine.scaler.scale(engine.state.output[Keys.LOSS]).backward()"

Here is my json code.
https://gist.github.com/katy070222/20a694f66c2676e003dc5e30308cf34f

Please help me.

**Comment by KumoLiu** [comment, marked as answer]
Hi @katy070222, if you want to change the output channel, you could do something like this:
```
import torch
from monai.networks.nets import SwinUNETR

model = SwinUNETR(
    img_size=(96, 96, 96),
    in_channels=1,
    out_channels=2,
    feature_size=48,
    use_checkpoint=True,
)

img = torch.rand(1, 1,96,96,96)
out = model(img)
```
Hope it can help you, thanks!

**Reply by katy070222** [reply]
Hi ,@KumoLiu ,thank you for your reply.

What if I can only modify the json file? 

I tried to modify like below,but it seems like not working 
![image](https://user-images.githubusercontent.com/119793057/216590469-33bb03a4-acd0-41ac-9672-5b4d7c2c4078.png)


And I also add "SpatialPadd" to size (96, 96, 96) in "random_transforms" before the "RandCropByPosNegLabeld".
![image](https://user-images.githubusercontent.com/119793057/216590567-0145f9ff-3b55-40d6-9743-52361d6ed4d4.png)

But it still stop at that.

Here is my train.json.
https://gist.github.com/katy070222/6aee8a63f22159741d98b140bc7ce79e
.

**Reply by KumoLiu** [reply]
Hi @katy070222, 
I can't reproduce your error. I modified the JSON file in the model zoo repo which make it can be used to run spleen data. Attach the JSON file for your reference. [train.txt](https://github.com/Project-MONAI/tutorials/files/10578583/train.txt)

Hope it can help you, thanks!

**Reply by KumoLiu** [reply]
Hi @katy070222, from the error message, I think your path is not set correctly. There are some possible reasons:
- You aren't under the correct path
- You didn't download the model

However, if you retrain the model, it doesn't need the weight.

You can download the model weight by using `python -m monai.bundle download --name swin_unetr_btcv_segmentation --version xxx --bundle_dir .`
Thanks!

**Reply by katy070222** [reply]
Hi ,@KumoLiu ,thank you for your reply.

yes. I just know this fault. Sorry,I would quickly modify it. I'm so sorry.

![image](https://user-images.githubusercontent.com/119793057/216611741-96f44dad-c542-45a7-b3af-37734ded69c3.png)

And,I got this wrong.

Because my input spatial_shape is "480, 640, 163" ,what should I do?

I have used "SpatialPadd' before the origin "random_transforms" ,and it work.
But I want to asking is it ok for my training? 

![image](https://user-images.githubusercontent.com/119793057/216612890-eddd64a8-86b2-4a38-9ba0-eb41d309e748.png)

**Reply by KumoLiu** [reply]
Hi @katy070222, if your input spatial shape is (480, 640, 163), it won't raise `The size of the proposed random crop ROI is larger than the image size.` this error, since your crop shape (96, 96, 96) is smaller than image shape.
Maybe you should check this `Spacingd` transform to set the proper pixdim.
Additionally, you can also set `allow_smaller=True` in `RandCropByPosNegLabel`, so that any smaller dimensions will be set to match the cropped size.
https://github.com/Project-MONAI/MONAI/blob/a77de4adff49037ffb24ea2cae217fac40302181/monai/transforms/croppad/array.py#L1002

Thanks!

**Reply by katy070222** [reply]
Hi ,@KumoLiu ,thank you for your reply.

I tried to do what you write,and it work!

Thank you very much for helping me a lot!
Without you ,maybe I would be trapped by those fault.

Thanks!
